controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -259,6 +268,41 @@ func (r *BareMetalHostReconciler) Reconcile(request ctrl.Request) (result ctrl.R
 	return
 }
 
+// Consume inspect.metal3.io/hardwaredetails when either
+// inspect.metal3.io=disabled or there are no existing HardwareDetails
+func (r *BareMetalHostReconciler) updateHardwareDetails(request ctrl.Request, host *metal3v1alpha1.BareMetalHost) (bool, error) {
+	reqLogger := r.Log.WithValues("baremetalhost", request.NamespacedName)
+	updated := false
+	if host.Status.HardwareDetails == nil || inspectionDisabled(host) {
+		objHardwareDetails, err := r.getHardwareDetailsFromAnnotation(host)
+		if err != nil {
+			reqLogger.Info("Error getting HardwareDetails from annotation")
+			return updated, err
```

Comment:
Should this error be wrapped with a message like what's logged in the line above to provide context?

Reply:
Yeah, I was wrapping the error returned from the function, but including this context makes sense, now updated, thanks!